test/ng/rootScopeSpec.js

```diff
@@ -212,10 +212,52 @@ describe('Scope', function() {
       expect($rootScope.$$watchersCount).toBe(2);
     }));
 
-    it('should not keep constant literals on the watch queue', inject(function($rootScope) {
+    it('should remove $watch of constant literals after initial digest', inject(function($rootScope) {
       $rootScope.$watch('[]', function() {});
       $rootScope.$watch('{}', function() {});
-      expect($rootScope.$$watchers.length).toEqual(2);
+      $rootScope.$watch('1', function() {});
+      $rootScope.$watch('"foo"', function() {});
+      expect($rootScope.$$watchers.length).not.toEqual(0);
+      $rootScope.$digest();
+
+      expect($rootScope.$$watchers.length).toEqual(0);
+    }));
+
+    it('should remove $watchCollection of constant literals after initial digest', inject(function($rootScope) {
```

Comment:
Ok, how about wrapping the constant related watch and watchCollection tests with another describe? Btw, what about watchGroup :D